### PR TITLE
Filter `filetype` of Wikimedia's images

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Part of #4717 by @obulat 

## Description

Adds a validation step for image file types in the Wikimedia provider DAG. I considered making this validation global (in the MediaStore class), but we must be really sure of the file types that will be accepted as valid on each media type (maintaining the `EXTENSIONS` dictionary up to date or validated in some way). So, for now, I limited the scope to Wikimedia images, which have been the most problematic.

## Testing Instructions

Run the Wikimedia DAG and check the filetypes in the upstream database.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog PRs) or the media properties generator (`ov just catalog/generate-docs media-props` for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py [hunk 1]
```diff
@@ -399,6 +399,24 @@ def test_extract_creator_info_handles_link_as_partial_text(wmc):
...
+        ("https://example.com/image.jpg", "image", "jpg"),
+        ("https://example.com/image.JpeG", "image", "jpeg"),
+        ("https://example.com/image.Png", "image", "png"),
+        ("https://example.com/image.GIF", "image", "gif"),
+        ("https://example.com/image.ogv", "image", None),
+        ("https://example.com/image.xyz", "image", None),
+        ("https://example.com/audio.mp3", "audio", "mp3"),
+        ("https://example.com/audio.ogg", "audio", "ogg"),
+        ("https://example.com/audio.WAV", "audio", "wav"),
+    ],
```
Comment (AetherUnbound) [inline catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py:414]: Nit: do you mind adding comments to separate these?
```suggestion
    [
        # Valid images
        ("https://example.com/image.jpg", "image", "jpg"),
        ("https://example.com/image.JpeG", "image", "jpeg"),
        ("https://example.com/image.Png", "image", "png"),
        ("https://example.com/image.GIF", "image", "gif"),
        # Invalid (for our sake) images
        ("https://example.com/image.ogv", "image", None),
        ("https://example.com/image.xyz", "image", None),
        # Valid audio
        ("https://example.com/audio.mp3", "audio", "mp3"),
        ("https://example.com/audio.ogg", "audio", "ogg"),
        ("https://example.com/audio.WAV", "audio", "wav"),
    ],
```

**Review by AetherUnbound — APPROVED** [review]
LGTM, seems reasonable to put this here! Your point about moving this to the `MediaStore` class is interesting...but I agree that it's worth targeting Wikimedia specifically in this case. Thanks for addressing this!

File: catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py (hunk 1 above)
Comment (krysal) [reply catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py]: Good notes 👍

**Comment by sarayourfriend** [comment]
@krysal this almost seems like something that should be implemented for all DAGs? Maybe at the store level? Wondering what the use case is for a provider ingestion DAG to _not_ check this, if `filetype IS NOT NULL` (i.e., it is known). @WordPress/openverse-catalog, would this warrant a more abstract approach to handle this check for all providers?

**Comment by obulat** [comment]
@sarayourfriend, we do extract URL extension to use as the file type in the `MediaStore` if the file type is `None`. However, we only return the file type if it's a valid type for the media type. So, for an `.ogv` file, this would return `None`, but would still use the file. This is why we do need to filter Wikimedia items in the script itself. For other providers, the URL might not have an extension, so returning `None` as the filetype should be continued.

For this PR, though, we should re-use the existing utility for extracting extensions from here: https://github.com/WordPress/openverse/blob/5e2e0e308a3f26cf87edc731287aaac8b17117d4/catalog/dags/common/extensions.py#L7

**Comment by sarayourfriend** [comment]
If that's true then how does this PR address the issue of OGV files present in the images table? It returns `None` if the filetype is ogv (if the extract filetype is not in the supported extensions it doesn't return the filetype).

Doesn't that mean the MediaStore would just see None for the filetype and then continue including the file (treating it like when the URL doesn't have an extension)? Wouldn't this change actually make the data _worse_ because now we won't even see when the `filetype` is OGV without doing a pattern match query on the URL?

I guess I don't understand how this is intended to validate anything. It filters metadata out of the final image row without affecting whether that image is kept. In these cases we ostensibly _know_ the filetype (at least as much as any other place where we know the filetype based on the URL extension) but we're just ignoring it. Why? What's the ultimate goal of this PR? @krysal's description explains it just as "validation", but I don't understand to what end, if we aren't filtering these results entirely out of ingestion by the time we see they are not of a supported filetype.

**Comment by obulat** [comment]
@sarayourfriend, I assumed that this PR returned `None` from `get_record_data` if the extracted extension is `None`. That's why I wrote a very long explanation of how Wikimedia script and `MediaStore` worked together.

But then I saw that this PR only extracts a valid extension from the URL, and does not drop the image if the extension is invalid.

@krysal, this PR should be updated to:
1. use the existing function for extracting extension (note: it needs to be updated to `.lower()` the URL, otherwise it won't work for capitalized extensions in some Wikimedia URLs)
2. return `None` from `get_record_data` if the extension could not be extracted. 

I think we should also log the URLs with the valid Wikimedia (`BITMAP` and `AUDIO`?) from which we could not extract a valid extension, to see what other extensions we might be missing.

**Comment by AetherUnbound** [comment]
> I assumed that this PR returned `None` from `get_record_data` if the extracted extension is `None`. 

Ahh, I had made the same assumption. I agree with Olga's suggested approach here.

**Comment by krysal** [comment]
I don't know how that "tiny" detail was overlooked 🙃 My bad! I created #4785 to follow up. @obulat, it uses an approach quite similar to what you describe, with improvements to the common function to get the file type. It needs to update more tests, but you all let me know if it's what is expected.

**Comment by sarayourfriend** [comment]
Should we revert this, considering it is potentially actively creating worse metadata for all new Wikimedia ingestion runs, rather than actually doing any filtering or augmenting metadata in a way we could use?

**Comment by krysal** [comment]
@sarayourfriend It's not doing anything much different that before so I wouldn't say it creating _worse_ metadata. If anything, it's detecting and logging when the DAG might be ingesting an unexpected type. In any case, it's being changed in #4785.

**Comment by sarayourfriend** [comment]
It is actively excluding the file type column for works we do not wish to include. To me that is worse metadata, we have less information to work from to fix our Wikimedia data than we did otherwise.

But fine, we will fix it in more significant ways in #4785 and #4717.

**Pull request merged**